src/net_processing.cpp

```diff
@@ -1621,7 +1624,7 @@ void PeerManagerImpl::BlockChecked(const CBlock& block, const BlockValidationSta
     else if (state.IsValid() &&
              !m_chainman.ActiveChainstate().IsInitialBlockDownload() &&
              mapBlocksInFlight.count(hash) == mapBlocksInFlight.size()) {
-        if (it != mapBlockSource.end()) {
+        if (it != mapBlockSource.end() && m_mempool) {
```

Comment:
why is the check needed here?

Reply:
I think there might be some more redundant `m_mempool` checks that might already be handled by a `m_tx_relay` check, I will find and remove the redundant checks.